File: truss-compiler/src/main/java/truss/compiler/preprocessor/IfDirective.java
package truss.compiler.preprocessor;

class IfDirective implements Directive {
    private final boolean expression;

    public IfDirective(boolean expression) {
        this.expression = expression;
    }

    public boolean isExpression() {
        return expression;
    }

    @Override
    public DirectiveKind getKind() {
        return DirectiveKind.IF;
    }
}

File: Truss.Compiler/PreProcessor/IfDirective.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Truss.Compiler.PreProcessor {
    internal class IfDirective : IDirective {
        public IfDirective(bool expression) {
            Expression = expression;
        }

        public bool Expression { get; private set; }

        public DirectiveKind Kind {
            get { return DirectiveKind.If; }
        }
    }
}
